src/components/filters/DateFilter/utils/DateConversions.ts

```diff
@@ -3,12 +3,38 @@ import * as moment from "moment";
 import isString = require("lodash/isString");
 import isDate = require("lodash/isDate");
 import { platformDateFormat } from "../../../../constants/Platform";
+import { isNegativeGTMTime, getTimezoneOffset } from "./OptionUtils";
 
-export const convertDateToPlatformDateString = (date: Date | undefined | null): string | undefined | null =>
-    isDate(date) ? moment(date).format(platformDateFormat) : date;
+export const convertDateToPlatformDateString = (date: Date | undefined | null): string | undefined | null => {
+    return isDate(date) ? moment(date).format(platformDateFormat) : date;
+};
 
 export const convertPlatformDateStringToDate = (
     platformDate: string | undefined | null,
-): Date | undefined | null => (isString(platformDate) ? moment(platformDate).toDate() : platformDate);
+): Date | undefined | null => {
+    if (!isString(platformDate)) {
+        return platformDate;
+    }
 
-export const createDateFromString = (dateString: string) => (dateString ? new Date(dateString) : undefined);
+    if (isNegativeGTMTime()) {
```

Comment:
What happens when the GMT offset is positive? Or why you are offsetting time by its timezone offset when it is negative, but not when it is positive?

Reply:
To remain the same date as platformDate string, I suppose two simple ways
- return new Date(platformDateString + "T00:00:00Z"): force platformDate as midnight time of UTC+0
- using the formula : timeInUTCInMilliSeconds=localTimeInMilliSeconds + (new Date().getTimezoneOffset())*60000 
var platformDateInUTC = moment(platformDateString).utc().toDate().getTime(); // midnight of at localTime
var offset = new Date().getTimezoneOffset()*60000; // offset UTC and Local Time
return new Date(platformDateInUTC -  offset);
What do you think @petrjanu  ?

Reply:
Okay, makes sense, but I'm so confused of this timezone stuff, that I cannot confirm if this is the correct way. I was commenting the code because I was curious what is this about :) Thank you for exaplanation.